woocommerce/woocommerce-android
# Convert AnalyticsEvent back to be an Enum

### Description
This PR converts the `AnalyticsEvent` class back to be an `enum`, this was done to avoid any performance hit from having a big number of `objects` and also because using an `enum` needs less code, and results in a much smaller bytecode.

See for more context: p1717697396465749-slack-CGPNUU63E

**I didn't make any changes to `WooPosAnalytics` itself.**

### Testing information
1. Launch the app.
2. Check logcat.
3. Confirm there are events tracked correctly by checking for lines that start with `Tracked`
